src/utils/ucx/meson.build

```
@@ -21,8 +21,15 @@ endif
 
 ucx_utils_inc_dirs = include_directories('.')
 
+ucx_utils_sources = ['ucx_utils.cpp', 'ucx_utils.h', 'config.h', 'config.cpp', 'rkey.cpp', 'rkey.h']
+
+# Add device_mem_list files only if UCX GPU Device API is available
+if ucx_gpu_device_api_available
+    ucx_utils_sources += ['device_mem_list.cpp', 'device_mem_list.h']
+endif
```

```suggestion
ucx_utils_sources = ['ucx_utils.cpp', 'ucx_utils.h', 'config.h', 'config.cpp', 'rkey.cpp', 'rkey.h', 'device_mem_list.cpp', 'device_mem_list.h']
```

I recommend controlling the functionality at one level: using ifdef in the code, or at the meson level by including various files. Since the file-level control approach is already being used, it is better to stick to this path in this case.